proto/gov/v1beta/msg.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+package heimdall.gov.v1beta;
+
+import "cosmos/base/v1beta1/coin.proto";
+import "gov/v1beta/gov.proto";
+import "heimdall/types/validator.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/maticnetwork/heimdall/x/gov/types";
+
+option (gogoproto.sizer_all)       = true;
+option (gogoproto.marshaler_all)   = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// Msg defines the bank Msg service.
+service Msg {
+    // SubmitProposal defines a method to create new proposal given a content.
+    rpc SubmitProposal(MsgSubmitProposal) returns (MsgSubmitProposalResponse);
+
+    // Vote defines a method to add a vote on a specific proposal.
+    rpc Vote(MsgVote) returns (MsgVoteResponse);
+
+    // Deposit defines a method to add deposit on a specific proposal.
+    rpc Deposit(MsgDeposit) returns (MsgDepositResponse);
+}
+
+// MsgSubmitProposal defines an sdk.Msg type that supports submitting arbitrary
+// proposal Content.
+message MsgSubmitProposal {
+    option (gogoproto.equal)           = false;
+    option (gogoproto.stringer)        = false;
+    option (gogoproto.goproto_getters) = false;
+
+    google.protobuf.Any content = 1
+        [(cosmos_proto.accepts_interface) = "Content"];
+    repeated cosmos.base.v1beta1.Coin initial_deposit = 2 [
+        (gogoproto.nullable)     = false,
+        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+        (gogoproto.moretags)     = "yaml:\"initial_deposit\""
+    ];
+    string                     proposer  = 3 [(gogoproto.casttype) =
+                             "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    heimdall.types.ValidatorID validator = 4;
+}
+
+// MsgSubmitProposalResponse defines the Msg/SubmitProposal response type.
+message MsgSubmitProposalResponse {
+    uint64 proposal_id = 1 [
+        (gogoproto.jsontag)  = "proposal_id",
+        (gogoproto.moretags) = "yaml:\"proposal_id\""
+    ];
+}
+
+// MsgVote defines a message to cast a vote.
+message MsgVote {
+    option (gogoproto.goproto_getters) = false;
+
+    uint64 proposal_id = 1 [
+        (gogoproto.jsontag)  = "proposal_id",
+        (gogoproto.moretags) = "yaml:\"proposal_id\""
+    ];
+    string                     voter     = 2 [(gogoproto.casttype) =
```

Comment:
We can remove casttype

Reply:
why we need to remove case type here?

Reply:
@temaniarpit27 because GRPC Gateway is not supporting cast types